### PR TITLE
Add AB test to test two-stage test bucketing

### PR DESCRIPTION
**Pull request by vlbee**
## What does this change?

Introduces an AB test to test two-stage bucketing in isolation. 

Frontend AB Test Switch Code: https://github.com/guardian/frontend/pull/25553

We require an implementation of two-stage bucketing for the long-running mandatory sign in gate trial.

#### How it should work
- During a bucketing window, new browsers are added to the test bucket based on their mvt_id cookie range but we also set a local storage key for the ab test under `"gu.ab.participations"` . 
- After the bucketing window, we stop setting the local storage key and check for its presence instead.  The ab test variant _should only run_ if the correct ab test is set in `"gu.ab.participations"`. This means that new browsers falling within the mvt_id bucket range should _**not**_ appear in the test, becuase the local storage key will not be set for them. 

#### How it is implemented
- This is implemented by a function called 'setOrUseParticipations'[ here](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/experiments/lib/ab-exclusions.ts)
- This function can be used within an AB Test Definition's Variant's `canRun` function 
- The bucketing is controlled by a boolean flag passed into  'setOrUseParticipations'

#### Data expectations

We expect to see new, unique browsers (represented by BrowserIds in the pageview table) entering the test and seeing the gate during the bucketing phase.

After the bucketing phase (running phase), we expect:
- Continued ab test component events firing for the unqiue browser ids from the bucketing phase. 
- No new browser ids bucketed in the test after the bucketing phase. 
- No ab test component events fired for NEW browser ids 

## Why?

We did not observe the expected data in the[ previous test run](https://github.com/guardian/dotcom-rendering/pull/5891), so are testing the local storage participation bucketing in isolation over a 48hr period. 

24hrs with flag set to true (bucketing)
24hrs with flag set to false (no bucketing)

## Local Testing Observations

**1st Set up - Ensure the setParticipationsFlag is true**
- Run branch - `make watch`
- Frontend PR - set the test's AB switch to "ON".
- Run Frontend PR client - `make dev` 
- Run Frontend PR server side `./sbt  project article run` 

1. Open new, clean browser window. 
2. Navigate to local Article page. Accept CMP cookies. 
3. Set 'GU_mvt_id' cookie in range eg. 897502
4. Refresh page. 
 
|   |   Expectation      | 
|----------|:-------------:|
| ab.participation |  newly set | 
| GU_mvt_id |   Set   | 
| Ophan tracking | correct abTestRegistration query param fired| 
| Gate |appears after a few refreshes |

5. Switch Participations Flag to false and reload DCR 
6. Delete `gu.ab.participation` from local storage
7. Refresh page

|    |    Expectation    | 
|----------|:-------------:|
| ab.participation |  not set | 
| GU_mvt_id |   Set   | 
| Ophan tracking | no abTestRegistration query param fired| 
| Gate | does not appear|

**2nd Set up - Ensure the setParticipationsFlag is False**

1. Open new, clean browser window. 
2. Navigate to local Article page. Accept CMP cookies. 
4. Set 'GU_mvt_id' cookie in range eg. 897502
5. Refresh page. 

|   |   Expectation      | 
|----------|:-------------:|
| ab.participation |  NOT set | 
| GU_mvt_id |   Set   | 
| Ophan tracking | no abTestRegistration query param fired| 
| Gate | does not appear after many refreshses |

### CODE REVIEW

File: dotcom-rendering/src/web/experiments/tests/sign-in-gate-mandatory-long-testrun.ts [hunk 1]
```diff
@@ -168,21 +17,20 @@ export const signInGateMandatoryLongTestRunEu: ABTest = {
 	successMeasure: 'Users sign in or create a Guardian account',
 	audienceCriteria:
 		'Global, 3rd article of the day, lower priority than consent banner, simple articles (not gallery, live etc.), not signed in, not shown after dismiss, not on help, info sections etc. Exclude iOS 9 and guardian-live-australia. Suppresses other banners, and appears over epics',
-	dataLinkNames: 'SignInGateMandatoryLongTestRunEu',
+	dataLinkNames: 'SignInGateMandatoryLongTestRunUk',
```
Comment (ioannakok) [inline dotcom-rendering/src/web/experiments/tests/sign-in-gate-mandatory-long-testrun.ts:20]: Should this be `SignInGateMandatoryLongTestRunBucketingRun`?

Comment (vlbee) [reply dotcom-rendering/src/web/experiments/tests/sign-in-gate-mandatory-long-testrun.ts]: yes! Thank you for spotting!

**Review by ioannakok — APPROVED** [review]
Looks good!

**Comment by vlbee** [comment]
I am adding additional logging to this test run to help debug

**Review by ioannakok — APPROVED** [review]
Looks great!

**Pull request merged**